### ISSUE TITLE
doesn't complain about empty lines

### ISSUE THREAD

**Issue by kzidane** [cs50/style50#41]
Should be considered bad style? cc @spencertiberi @Erin-c

```
/* foo.c */
#include <stdio.h>

int main(void)
{
    if (1)
    {
        printf("true\n");


    }
}
```

```
$ style50 foo.c
Looks good!
But consider adding more comments!
```

**Comment by cmlsharp** [comment]
`astyle` has an option for deleting empty lines, but doing so will delete **all** empty lines which is far too strict (and potentially bad style anyway).

**Comment by cmlsharp** [comment]
Trying to implement this ourselves, outside the purview of `astyle` doesn't seem particularly promising either, as this would require somewhat of a semantic understanding of the code. The posted example is obviously bad style, but is e.g.

```c
int x = 4;

for (size_t i = 0; i < x; i++)
{
    // stuff
}
```

**Comment by cmlsharp** [comment]
Hmm... actually that last bit is a bad example since `--break-blocks` will re-add the blank line there anyway. If it is the case that extraneous blank lines (not surrounding `if/while/for/etc.` blocks) are always bad style then we could add this option, but I'm still a bit worried that there will be too many false positives.

Will explore more.

**Comment by cmlsharp** [comment]
`--delete-empty-lines` is indeed too restrictive. It removes blank lines from around comments so 

```c
// Does thing
do_thing();

// Does other thing
do_other_thing();
```
gets squished into

```c
// Does thing
do_thing();
// Does other thing
do_other_thing();
 ```
which isn't necessarily better style.

As such, I don't think this is readily fixable without switching to another styling tool.

**Comment by cmlsharp** [comment]
Oops
